integration-tests/admission/admission_test.go

```diff
@@ -0,0 +1,291 @@
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/openebs/maya/integration-tests/artifacts"
+	k8s "github.com/openebs/maya/pkg/client/k8s/v1alpha1"
+	cv "github.com/openebs/maya/pkg/cstorvolume/v1alpha1"
+	pvc "github.com/openebs/maya/pkg/kubernetes/persistentvolumeclaim/v1alpha1"
+	validatehook "github.com/openebs/maya/pkg/kubernetes/webhook/validate/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	cStorPVC artifacts.Artifact = `
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: cstor-source-volume
+  namespace: default
+  labels:
+    name: cstor-test
+spec:
+  storageClassName: openebs-cstor-class
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: "2G"
+`
+	// singleReplicaSC holds the yaml spec
+	// for pool with single replica
+	singleReplicaSC artifacts.Artifact = `
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: openebs-cstor-class
+  annotations:
+    cas.openebs.io/config: |
+      - name: StoragePoolClaim
+        value: "cstor-sparse-pool"
+      - name: ReplicaCount
+        value: "1"
+    openebs.io/cas-type: cstor
+provisioner: openebs.io/provisioner-iscsi
+reclaimPolicy: Delete
+`
+	// clonePVCYaml holds the yaml spec
+	// for clone persistentvolumeclaim
+	clonePVCYaml artifacts.Artifact = `
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: test-snap-claim
+  namespace: default
+  labels:
+    name: test-snap-claim
+  annotations:
+    snapshot.alpha.kubernetes.io/snapshot: snapshot-cstor
+spec:
+  storageClassName: openebs-snapshot-promoter
+  accessModes: [ "ReadWriteOnce" ]
+  resources:
+    requests:
+      storage: 2G`
+	// cstorSnapshotYaml holds the yaml spec
+	// for volume snapshot
+	cstorSnapshotYaml artifacts.Artifact = `
+apiVersion: volumesnapshot.external-storage.k8s.io/v1
+kind: VolumeSnapshot
+metadata:
+  name: snapshot-cstor
+  namespace: default
+spec:
+  persistentVolumeClaimName: cstor-source-volume
+`
+)
+
+var _ = Describe("[single-node] [cstor] AdmissionWebhook", func() {
+	BeforeEach(func() {
+		// Extracting storageclass artifacts unstructured
+		SCUnst, err := artifacts.GetArtifactUnstructured(singleReplicaSC)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		// Apply  single replica storageclass
+		cu := k8s.CreateOrUpdate(
+			k8s.GroupVersionResourceFromGVK(SCUnst),
+			SCUnst.GetNamespace(),
+		)
+
+		_, err = cu.Apply(SCUnst)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		// Extracting PVC artifacts unstructured
+		PVCUnst, err := artifacts.GetArtifactUnstructured(cStorPVC)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		// Extracting pvc namespace
+		PVCNamespace := PVCUnst.GetNamespace()
+
+		// Webhook stuffs
+		_, err = validatehook.KubeClient().List(metav1.ListOptions{})
+		if errors.IsNotFound(err) {
+			Fail(fmt.Sprintf("dynamic configuration of webhooks requires the admissionregistration.k8s.io group to be enabled: %v", err))
```

Comment:
Why is this being done here?
This should be part of suite_test.go?

Reply:
Why are we introducing new concept i.e. `Fail(...)`

Reply:
There is no new concept  IMO . Fail(..) is just a function of `gomega` , which we are already using it.
Anyways i will change this

Reply:
Done